architecture/core_concepts/image_streams.adoc

```diff
@@ -0,0 +1,614 @@
+the second position, and so forth. This allows for easy rollbacks to make tags point to
+historical images again.
+
+Image stream tags can be _reference tags_ or _version-specific tags_. 
+
+* Version-specific tags point to a particular version of an image, such as Python 3.5.
+
+* Reference tags follow another image stream tag and could be updated in the future to change which image they follow, much like a symlink. Note that these new levels are not guaranteed to be backwards compatible. 
++
+For example, the `latest` tags that ship with OpenShift are _reference-style_tags. This means consumers of the `latest` tag will be updated to the newest level of the framework provided by the image when a new level becomes available. A `latest tag to v3.6 could be changed to v3.7 at any time.
+
+//Ben: elaborate on what it means to "push an image without properly tagging it" 
+
+The  "latest" image stream tags specifically, are reference tags which 
+
+However users can create their own image stream tags named anything they want. And those image stream tags might or might not be reference-style tags.
+
+
+The image stream tag is composed of two parts separated by a colon: `<image stream
+name>:<tag>`. 
+
+For example, the image stream tag referring to the image
+`sha256:47463d94eb5c049b2d23b03a9530bf944f8f967a0fe79147dd6b9135bf7dd13d` in
+the xref:image-stream-object-definition[example above] would be
+`origin-ruby-sample:latest`.
+
+.Image Stream Tag with Two Images in its History
+====
+[source,yaml]
+----
+  tags:
+  - items:
+    - created: 2016-03-02T10:15:09Z
+      dockerImageReference: 172.30.56.218:5000/test/origin-ruby-sample@sha256:909de62d1f609a717ec433cc25ca5cf00941545c83a01fb31527771e1fab3fc5
+      generation: 2
+      image: sha256:909de62d1f609a717ec433cc25ca5cf00941545c83a01fb31527771e1fab3fc5
+    - created: 2016-01-29T13:40:11Z
+      dockerImageReference: 172.30.56.218:5000/test/origin-ruby-sample@sha256:47463d94eb5c049b2d23b03a9530bf944f8f967a0fe79147dd6b9135bf7dd13d
+      generation: 1
+      image: sha256:47463d94eb5c049b2d23b03a9530bf944f8f967a0fe79147dd6b9135bf7dd13d
+    tag: latest
+----
+====
+
+
+[[image-stream-triggers]]
+== Image Stream Triggers
+
+Image change triggers allow your build or deployment to be automatically invoked when a new version of an upstream image is available.
+
+//from FAQ
+For example, xref:../dev_guide/builds/triggering_builds.adoc#image-change-triggers[Builds] and 
+xref:../dev_guide/deployments/basic_deployment_operations.adoc#image-change-trigger[Deployments] can be automatically started when an Image Stream Tag is modified. 
+This is achieved by monitoring that particular Image Stream Tag and notifying the controller (the Build or Deployment) when a change was detected.
+
+include::dev_guide/deployments/basic_deployment_operations.adoc[tag=image-change-trig]
+
+[[image-stream-mappings]]
+== Image Stream Mappings
```

Comment:
Can this be marked as an advanced topic? Regular users don't need to bother with those implementation details.

Reply:
Added a note below.
"Configuring image stream mappings is an advanced feature."
@soltysh 
Do we want to define when a user would need to create an image stream mapping?